gammapy/data/observations.py

```diff
@@ -744,6 +744,19 @@ def from_stack(cls, observations_list):
         obs = itertools.chain(*observations_list)
         return cls(list(obs))
 
+    def iter(self, idx=None):
```

Comment:
Maybe call this `generator()` not to confuse with the `__iter__()` method?

I would prefer no to pass the slice as argument and rely on `Observations[slice]` instead, but I understand that the latter returns a `list` and not an `Observations`. Maybe this is an issue that should be corrected.

Reply:
Done. For the slice, see #4815.